File: restaurant_bar_app_server/models/mesero.js
const { Schema, model } = require('mongoose');

const MeseroSchema = Schema({

    usuario: {
        type: String,
        required: true
    },

    nombre: {
        type: String,
        required: true
    },
    apellidos: {
        type: String,
        required: true
    },
    type: {
        type: String,
        required: true
    },

});

MeseroSchema.method('toJSON', function () {
    const { __v, ...object } = this.toObject();
    return object;
})



module.exports = model('Mesero', MeseroSchema);

File: restaurant_bar_app/lib/models/mesero.dart
class Mesero {
  Mesero({
    this.id,
    this.usuario,
    this.nombre,
    this.apellidos,
    this.type,
  });

  String id;
  String usuario;
  String nombre;
  String apellidos;
  String type;

  factory Mesero.fromJson(Map<String, dynamic> json) => Mesero(
        id: json["_id"],
        usuario: json["usuario"],
        nombre: json["nombre"],
        apellidos: json["apellidos"],
        type: json["type"],
      );

  Map<String, dynamic> toJson() => {
        "_id": id,
        "usuario": usuario,
        "nombre": nombre,
        "apellidos": apellidos,
        "type": type,
      };
}
